docs/fspiop-rest-v1.0-OpenAPI-implementation.yaml

```diff
@@ -0,0 +1,4664 @@
+      - TRANSFER Used for performing a P2P (Peer to Peer, or Consumer to Consumer) transaction.
+      - PAYMENT Usually used for performing a transaction from a Consumer to a Merchant or Organization, but could also be for a B2B (Business to Business) payment. The transaction could be online for a purchase in an Internet store, in a physical store where both the Consumer and Business User are present, a bill payment, a donation, and so on.
+      - REFUND Used for performing a refund of transaction. 
+  TransactionState:
+    title: TransactionStateEnum
+    type: string
+    enum:
+    - RECEIVED
+    - PENDING
+    - COMPLETED
+    - REJECTED
+    description: Below are the allowed values for the enumeration
+      - RECEIVED Payee FSP has received the transaction from the Payer FSP.
+      - PENDING Payee FSP has validated the transaction.
+      - COMPLETED Payee FSP has successfully performed the transaction.
+      - REJECTED Payee FSP has failed to perform the transaction.
+  TransactionSubScenario:
+    title: TransactionSubScenario
+    type: string
+    pattern: ^[A-Z_]{1,32}$
+    description: Possible sub-scenario, defined locally within the scheme (UndefinedEnum Type).
+  TransferState:
+    title: TransferStateEnum
+    type: string
+    enum:
+    - RECEIVED
+    - RESERVED
+    - COMMITTED
+    - ABORTED
+    description: Below are the allowed values for the enumeration
+      - RECEIVED Next ledger has received the transfer.
+      - RESERVED Next ledger has reserved the transfer.
+      - COMMITTED Next ledger has successfully performed the transfer.
+      - ABORTED Next ledger has aborted the transfer due a rejection or failure to perform the transfer.
+  UndefinedEnum:
+    title: UndefinedEnum
+    type: string
+    pattern: ^[A-Z_]{1,32}$
+    description: The API data type UndefinedEnum is a JSON String consisting of 1 to 32 uppercase characters including an underscore character (_).
+  U2F:
+    title: U2F
+    type: string
+    minLength: 1
+    maxLength: 64
+    description: U2F challenge-response, where payer FSP verifies if the response provided by end-user device matches the previously registered key.
+
+  #Complex Types
+  AuthenticationInfo:
+    title: AuthenticationInfo
+    type: object
+    description: Data model for the complex type AuthenticationInfo
+    properties:
+      authentication:
+        $ref: '#/definitions/AuthenticationType'
+        description: Type of authentication.
+      authenticationValue:
+        $ref: '#/definitions/AuthenticationValue'
+        description: Authentication value.
+    required:
+    - authentication
+    - authenticationValue
+
+  AuthenticationValue:
+    title: AuthenticationValue
+    type: object
+    description: Data model for the complex type AuthenticationValue
+    properties:
+      authenticationVal:
+        $ref: '#/definitions/AuthenticationVal'
```

Comment:
I'm stumped in coming up with a better name for this... any ideas @MichaelJBRichards @jgeewax ?

Reply:
Challenge? PinValue?

Reply:
As per review comment AuthenticationVal renamed to PinValue